unity/Assets/UnitTests/TestThirdPartyCameraAndMainCamera.cs

```diff
@@ -312,5 +312,329 @@ public IEnumerator TestUpdateMainCamera()
+                metadata.agentPositionRelativeCameraRotation.x,
+                30.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.agentPositionRelativeCameraRotation.y,
+                10.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.agentPositionRelativeCameraRotation.z,
+                12.0000000000f
+            );
+            Assert.AreEqual(result, true);
+        }
+
+        //test third party camera metadata
+        [UnityTest]
+        public IEnumerator TestThirdPartyCameraMetadataReturn()
+        {
+            bool result = false;
+
+            Dictionary<string, object> action = new Dictionary<string, object>();
+
+            action["action"] = "Initialize";
+            action["fieldOfView"] = 90f;
+            action["snapToGrid"] = true;
+            yield return step(action);
+
+            action.Clear();
+
+            action["action"] = "AddThirdPartyCamera";
+            action["position"] = new Vector3(3, 2, 1);
+            action["rotation"] = new Vector3(10, 20, 30);
+            action["parent"] = "world";
+            yield return step(action);
+
+            MetadataWrapper metadata = getLastActionMetadata();
+
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraPosition.x,
+                3.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraPosition.y,
+                2.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraPosition.z,
+                1.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraRotation.x,
+                10.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraRotation.y,
+                20.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraRotation.z,
+                30.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            Assert.AreEqual(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraPosition,
+                null
+            );
+            Assert.AreEqual(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraRotation,
```

Comment:
Change variable name to `agentRotationRelativeThirdPartyCameraRotation`

Reply:
updated in ef3bfb20c08ec9a7a991b23d7e9008af3c72d132